src/indexer.c

```diff
@@ -151,6 +151,12 @@ int git_indexer_new(
 	if (fd != -1)
 		p_close(fd);
 
+	if (git_buf_is_allocated(&tmp_path))
+		(void)p_unlink(git_buf_cstr(&tmp_path));
```

Comment:
We do not cast unused return values to `void`.  There are a few other places in here that should be removed also.